scripts/run-monitor-ci-tests.bash

```diff
@@ -4,87 +4,109 @@ set -eu -o pipefail
+		pipeline_ui_sha=$(echo ${config} | jq '.compiled' | grep -o 'export UI_SHA=[^\]*' | grep -v 'export UI_SHA=${LATEST_SHA}' | head -1 | sed 's/=/\n/g' | tail -1 || true)
+
+		if [[ "${SHA}" == "${pipeline_ui_sha}" ]]; then
+			# check if this pipeline's 'build' workflow is passing
+			workflows=$(curl -s --request GET \
+				--url "https://circleci.com/api/v2/pipeline/${pipeline_id}/workflow" \
+				--header "Circle-Token: ${API_KEY}" \
+				--header 'content-type: application/json' \
+				--header 'Accept: application/json')
+
+			number_build_success_workflows=$(echo ${workflows} | jq '.items | map(select(.name == "build" and .status == "success")) | length')
+			if [ $number_build_success_workflows -gt 0 ]; then
+				# we've found a successful run
+				found_passing_pipeline=1
+				break
+			fi
 		fi
+	done
+
+	# terminate early if we found a passing pipeline for this SHA
+	if [ $found_passing_pipeline -eq 1 ]; then
+		printf "\nSUCCESS: Found a passing monitor-ci pipeline for this SHA, will not re-run these tests\n"
+		exit 0
+	else
+		printf "\nno passing monitor-ci pipelines found for this SHA, starting a new one\n"
 	fi
-done
 
-# terminate early if we found a passing pipeline for this SHA
-if [ $found_passing_pipeline -eq 1 ]; then
-	printf "\nSUCCESS: Found a passing monitor-ci pipeline for this SHA, will not re-run these tests\n"
-	exit 0
+	# start the monitor-ci pipeline from the UI repo
+	DEPLOY_PROD=false
+	if [[ "${UI_BRANCH}" == "master" ]]; then
+		DEPLOY_PROD=false # TODO: change this to true when we're ready to depend on this script
+	fi
+	printf "\nstarting monitor-ci pipeline targeting monitor-ci branch ${MONITOR_CI_BRANCH}, UI branch ${UI_BRANCH} and using UI SHA ${SHA}\n"
+	pipeline=$(curl -s --fail --request POST \
+		--url https://circleci.com/api/v2/project/gh/influxdata/monitor-ci/pipeline \
+		--header "Circle-Token: ${API_KEY}" \
+		--header 'content-type: application/json' \
+		--header 'Accept: application/json'    \
+		--data "{\"branch\":\"${MONITOR_CI_BRANCH}\", \"parameters\":{ \"ui-sha\":\"${SHA}\", \"ui-branch\":\"${UI_BRANCH}\", \"ui-pull-request\":\"${PULL_REQUEST}\", \"deploy-prod\":${DEPLOY_PROD}}}")
 else
-	printf "\nno passing monitor-ci pipelines found for this SHA, starting a new one\n"
-fi
-
-# start the monitor-ci pipeline
-DEPLOY_PROD=false
-if [[ "${UI_BRANCH}" == "master" ]]; then
-	DEPLOY_PROD=false # TODO: change this to true when we're ready to depend on this script
+	# start the monitor-ci pipeline from the influxdb repo
+	printf "\nstarting monitor-ci pipeline targeting monitor-ci branch ${MONITOR_CI_BRANCH} using OSS SHA ${OSS_SHA}\n"
+	pipeline=$(curl -s --fail --request POST \
+		--url https://circleci.com/api/v2/project/gh/influxdata/monitor-ci/pipeline \
+		--header "Circle-Token: ${API_KEY}" \
+		--header 'content-type: application/json' \
+		--header 'Accept: application/json'    \
+		--data "{\"branch\":\"${MONITOR_CI_BRANCH}\", \"parameters\":{ \"oss-sha\":\"${OSS_SHA}\" }}")
```

Comment:
could we simplify this if/else by having one curl request outside the if/else and just pass different objects to --data? Then the else block would just be creating the data object for OSS

Reply:
Good idea! I pushed a change for this, it does simplify things a bit.